lib/puppet/util/platform.rb

```diff
@@ -17,6 +17,29 @@ def default_paths
         %w{/usr/sbin /sbin}
       end
       module_function :default_paths
+
+      def fips_enabled?
+        # We currently only plan to support FIPS on RHEL7
+        # so we try to eliminate anything un-supported using
+        # what we have available in lieu of not being able to specifically
+        # check for it.
+        return false if Puppet::Util::Platform.windows?
+
+        # REMIND - How to check if we are on posix+linux system to avoid
+        # getting surprised on any odd-ball system that might not entertain
+        # the below file query.
+
+        fips_status_file = '/proc/sys/crypto/fips_enabled'
+
+        if File.exist?(fips_status_file) && File.size(fips_status_file) > 1 &&
+           File.open(fips_status_file, &:readline)[0].chr == '1'
+          true
+        else
+          false
+        end
```

Comment:
The `fips_enabled` file has a 0-length, so this check is no good:

```
[root@wwxwbjmbdegguig ~]# ls -la /proc/sys/crypto/fips_enabled
-r--r--r-- 1 root root 0 Jan 22 03:15 /proc/sys/crypto/fips_enabled
```

I would simplify this by just trying to read at most 1 byte from the file.

```ruby
begin
  File.read(fips_status_file, 1) == '1'
rescue
  false
end
```

Reply:
rspec is tripping on the above use for ruby 1.9.3...have restarted it just to see if it is transient but based on the error does not look like. May need to go either with the older rev or something different.